packages/jest-circus/src/utils.ts

```diff
@@ -230,6 +246,9 @@ export const callAsyncCircusFn = (
       timeoutID.unref && timeoutID.unref();
       clearTimeout(timeoutID);
       throw error;
+    })
+    .finally(() => {
+      completed = true;
```

Comment:
not too happy about this, but we don't have access to the `test` definition, and the promise has already been resolved. Might make sense to dispatch some "suite failed" event or something like that?

@aaronabramov thoughts?